module/data/item/templates/activated-effect.mjs

```diff
@@ -0,0 +1,106 @@
+        value: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.Duration"}),
+        units: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.DurationType"})
+      }, {label: "DND5E.Duration"}),
+      target: new foundry.data.fields.SchemaField({
+        value: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.TargetValue"}),
+        width: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.TargetWidth"}),
+        units: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.TargetUnits"}),
+        type: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.TargetType"})
+      }, {label: "DND5E.Target"}),
+      range: new foundry.data.fields.SchemaField({
+        value: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.RangeNormal"}),
+        long: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.RangeLong"}),
+        units: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.RangeUnits"})
+      }, {label: "DND5E.Range"}),
+      uses: new foundry.data.fields.SchemaField({
+        value: new foundry.data.fields.NumberField({required: true, min: 0, label: "DND5E.LimitedUsesAvailable"}),
+        max: new FormulaField({required: true, deterministic: true, label: "DND5E.LimitedUsesMax"}),
+        per: new foundry.data.fields.StringField({
+          required: true, blank: false, nullable: true, initial: null, label: "DND5E.LimitedUsesPer"
+        }),
+        recovery: new FormulaField({required: true, label: "DND5E.RecoveryFormula"})
+      }, {label: "DND5E.LimitedUses"}),
+      consume: new foundry.data.fields.SchemaField({
+        type: new foundry.data.fields.StringField({required: true, blank: true, label: "DND5E.ConsumeType"}),
+        target: new foundry.data.fields.StringField({
+          required: true, blank: false, nullable: true, initial: null, label: "DND5E.ConsumeTarget"
+        }),
+        amount: new foundry.data.fields.NumberField({required: true, label: "DND5E.ConsumeAmount"})
+      }, {label: "DND5E.ConsumeTitle"})
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static migrateData(source) {
+    this.migrateMaxUses(source);
+    this.migrateRanges(source);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Ensure a 0 in max uses is converted to an empty string rather than "0".
+   * @param {object} source  The candidate source data from which the model will be constructed.
+   */
+  static migrateMaxUses(source) {
+    if ( (source.uses?.max === 0) || (source.uses?.max === "0") ) source.uses.max = "";
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fix issue with some imported range data that uses the format "100/400" in the range field,
```

Comment:
Imported from where? Is this actually a migration from a format we used in the system previously?

Reply:
This wasn't used in the system, but I encountered this format in one of my older worlds with content imported from D&D Beyond. I figured it would be good to handle cases like this for the moment to ensure people don't lose items when we add these models.

Reply:
What currently happens to that data now that we changed those inputs on the sheet to `number` inputs? It's just lost?

Reply:
If you open the sheet it is set to `null`. We could just change the migrate to discard this information if you think this log is too specific, but we need something to prevent these items from erroring out.

Reply:
It's fine, I don't think there's any problem with leaving this in. I'm a little wary of setting a precedent for migrating module-specific data, but if this was a string field at some point then it's plausible a user might enter such data themselves.